sample-compose/app/src/androidTest/java/co/nimblehq/sample/compose/test/CoroutineTestRule.kt

```diff
@@ -0,0 +1,32 @@
+package co.nimblehq.sample.compose.test
+
+import co.nimblehq.sample.compose.util.DispatchersProvider
+import kotlinx.coroutines.*
+import kotlinx.coroutines.test.*
+import org.junit.rules.TestWatcher
+import org.junit.runner.Description
+
+@OptIn(ExperimentalCoroutinesApi::class)
+class CoroutineTestRule(
+    private val testDispatcher: TestDispatcher = UnconfinedTestDispatcher()
+) : TestWatcher() {
+
+    override fun starting(description: Description?) {
```

Comment:
@Wadeewee Sorry, however, I don't get that warning on my side. Looking deeper into the implementation of `TestWatcher`, it's a java class so `Description` is nullable because it does not come with `@NotNull` annotation. 🤔 

![image](https://user-images.githubusercontent.com/16315358/218327113-70304c1e-0ed5-47e1-bcdb-ef241610e046.png)


So in Kotlin, the default overriding will be like this:

![image](https://user-images.githubusercontent.com/16315358/218327154-020a8ac4-97b7-4274-9b41-b02675479c3c.png)

Could you please re-check again?

Reply:
@luongvo Checked on my end, I have the same experience as @Wadeewee 🤔 

When overriding `starting()`, it shows the tip as `description: Description!`
<img width="425" alt="Screenshot 2566-02-13 at 15 57 22" src="https://user-images.githubusercontent.com/53168251/218414322-6cc033db-3063-42ca-a54d-dbba61fe1697.png">

And from inside `TestWatcher.java`, it seems to add a `@NotNull` annotation after the fact on the argument 🤔 
<img width="518" alt="Screenshot 2566-02-13 at 15 57 31" src="https://user-images.githubusercontent.com/53168251/218414593-63926042-a585-436d-b2a8-b32e0cdf005b.png">

Perhaps there is some mismatch between our setups or something? What do you think? 👀

Reply:
@ryan-conway @Wadeewee [This one](https://github.com/junit-team/junit4) is the source code of the latest version of [`JUnit 4.13.2`](https://github.com/junit-team/junit4/releases/tag/r4.13.2). The [TestWatcher.java](https://github.com/junit-team/junit4/blob/main/src/main/java/org/junit/rules/TestWatcher.java) class contains 12 functions with 1 deprecated `skipped`.

Looking at my version, it's totally the same, without the `@NotNull` annotation implemented 🤔 

![image](https://user-images.githubusercontent.com/16315358/218429306-b3c2b6c0-7c34-41af-a6cb-51cb3fc9de73.png)

While your version has only 6 functions without any deprecated ones. I think your versions are not the expected one (JUnit 4.13.2), please re-check.

![image](https://user-images.githubusercontent.com/16315358/218429350-7815f78b-1ce7-48fc-8796-d06942b0b472.png)

Reply:
@Tuubz Could you please check this issue on your end? 💭

Reply:
Slack discussion thread here: https://nimble-co.slack.com/archives/G41UCTGHL/p1676284519781699 👋